src/Common/AzurePSCmdlet.cs

```diff
@@ -515,7 +521,13 @@ protected new void WriteError(ErrorRecord errorRecord)
 
         protected new void ThrowTerminatingError(ErrorRecord errorRecord)
         {
-            FlushDebugMessages(IsDataCollectionAllowed());
+            FlushDebugMessages();
+            //Use DisableErrorRecordsPersistence as opt-out for now, will replace it with EnableErrorRecordsPersistence as opt-in at next major release (November 2023)
+            if ((!AzureSession.Instance.TryGetComponent<IConfigManager>(nameof(IConfigManager), out var configManager)
+                || !configManager.GetConfigValue<bool>(ConfigKeysForCommon.DisableErrorRecordsPersistence)) && IsDataCollectionAllowed())
```

Comment:
Can this be simplified?

Reply:
do you mean mark the state of the config somewhere as a flag?